geth/transactions/factory.go

```diff
@@ -0,0 +1,188 @@
+package transactions
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/rpc"
+	"github.com/status-im/status-go/sign"
+)
+
+const (
+	// sendTransactionTimeout defines how many seconds to wait before returning result in sentTransaction().
+	sendTransactionTimeout = 300 * time.Second
+	rpcCallTimeout         = time.Minute
+
+	defaultGas = 90000
+)
+
+// Factory validates, signs and propagates transactions to the Ethereum network.
+type Factory struct {
```

Comment:
I'm afraid the word "Factory" has a very strong association with Factory pattern from class-based programming and does not really represent the purpose of this type.
As much as `-Manager` is bad, it's still better describes what this abstraction is for.

Reply:
I don't know. This class doesn't manage anything, it creates and propagates the transactions to the Ethereum network... But I think you're right, Factory is a worse of a name.

Reply:
`Transactor` 🤖

Reply:
Yeah, that might make sense. `Transactor#SendTransaction` and `Transaction#validateAndPropagate` make sense to me.